plugins/test/unit/plugins/importers/test_steps.py

```diff
@@ -355,3 +373,263 @@ def test_link_error(self, fake_link, fake_islink, fake_readlink):
+
+
+class TestClean(TestCase):
+
+    def test_init(self):
+        step = Clean()
+        self.assertEqual(step.step_id, constants.IMPORT_STEP_CLEAN)
+        self.assertTrue(step.description is not None)
+
+    @patch(MODULE + '.lib')
+    def test_process_main(self, fake_lib):
+        path = 'root/path-123'
+        repo_id = 'repo-123'
+
+        # test
+        step = Clean()
+        step.parent = Mock(storage_path=path, repo_id=repo_id)
+        step.process_main()
+
+        # validation
+        fake_lib.Repository.assert_called_once_with(path)
+        fake_lib.Remote.assert_called_once_with(repo_id, fake_lib.Repository.return_value)
+        fake_lib.Remote.return_value.delete.assert_called_once_with()
+
+    @patch(MODULE + '.lib')
+    def test_process_main_exception(self, fake_lib):
+        path = 'root/path-123'
+        importer_id = 'importer-xyz'
+
+        fake_lib.LibError = LibError
+        fake_lib.Remote.return_value.delete.side_effect = LibError
+
+        # test
+        try:
+            step = Clean()
+            step.parent = Mock(storage_path=path, importer_id=importer_id)
+            step.process_main()
+            self.assertTrue(False, msg='Delete remote exception expected')
+        except PulpCodedException, pe:
+            self.assertEqual(pe.error_code, errors.OST0003)
+
+
+class TestRemote(TestCase):
+
+    def test_init(self):
+        step = Mock()
+        repository = Mock()
+        remote = Remote(step, repository)
+        self.assertEqual(remote.step, step)
+        self.assertEqual(remote.repository, repository)
+
+    def test_url(self):
+        step = Mock()
+        step.parent = Mock(feed_url='http://')
+        remote = Remote(step, None)
+        self.assertEqual(remote.url, step.parent.feed_url)
+
+    def test_remote_id(self):
+        step = Mock()
+        step.parent = Mock(repo_id='123')
+        remote = Remote(step, None)
+        self.assertEqual(remote.remote_id, step.parent.repo_id)
+
+    def test_working_dir(self):
+        step = Mock()
+        remote = Remote(step, None)
+        self.assertEqual(remote.working_dir, step.get_working_dir.return_value)
+
+    def test_config(self):
+        step = Mock()
+        remote = Remote(step, None)
+        self.assertEqual(remote.config, step.get_config.return_value)
+
+    @patch('__builtin__.open')
+    def test_ssl_key_path(self, fake_open):
+        key = 'test-key'
+        config = {
+            importer_constants.KEY_SSL_CLIENT_KEY: key
+        }
+        working_dir = '/tmp/test'
+        step = Mock()
+        step.get_config.return_value = config
+        step.get_working_dir.return_value = working_dir
+        fp = Mock(__enter__=Mock(), __exit__=Mock())
+        fp.__enter__.return_value = fp
+        fake_open.return_value = fp
+
+        # test
+        remote = Remote(step, None)
+        path = remote.ssl_key_path
+
+        # validation
+        expected_path = os.path.join(working_dir, 'key.pem')
```

Comment:
Perhaps make an assertion about the permissions and ownership of the private key?